### PR TITLE
initial creation of container orchestration docs

### PR DESCRIPTION
**Pull request by GregoryIvo**
initial creation of container orchestration provider docs.
image

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to inderstand this section can be skipped.

**Screenshots:** If applicable, add screenshots to help explain your solution

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

**Comment by MMaiero** [comment]
@marcellorinaldo Can you please have a review?

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+---
+layout: page
+title:  "Container Orchestration Provider"
+categories: [builtin]
+---
+
+# What is the Container Orchestration Provider?
+The Container Orchestration Provider allows Kura to manage Docker. With this tool you can arbitrarily pull and deploy containerized software packages and run them on your gateway. This Provider allows the user to create, configure, start, and stop containers all from the browser. The bundle will also restart containers, if the gateway is restarted.
+
+The Container Orchestration service is composed by two bundles, one that exposes APIs for container management and one that implements those APIs. This API is exposed so that you can leverage it to implement containerization in your own Kura plugins.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: is composed **by** -> is composed **of**

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+***
+
+# How to use the Container Orchestration Provider
+
+## Before Starting
+For this bundle to function appropriately, the gateway must have docker installed and the docker daemon must be started and enabled.
+
+## Starting the Service
+
+To use this service select the **Docker-API** option located in the **Services** area. The docker-api service provides the following parameters: **Enabled**--activates the service when set to true, and **Docker Host URL**--provides a string which tells the service where to find the docker engine (best left to the default value). Optionally a user can provide a Repository/Registry URL and respective credentials so that containers will be pulled from a alternative source such as AWS-ECR.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: **Docker Host URL**--provides a string **which** tells -> **Docker Host URL**--provides a string **that** tells

Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: pulled from **a** alternative source -> pulled from **an** alternative source

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+Containers may be configured using the following fields:
+
+- **Enabled** - When true, api will create the described container. When false api will not create the container, or will destroy the container if already running.
+  
+- **Image Name** - Describes the docker image that will be used to create the container. Remember to ensure that the selected image supports the architecture of the host machine, or else the container will not be able to start.
+  
+- **Image Tag** - Describes the version of the docker image that will be used to create the container.
+  
+- **Internal Ports** - This field accepts a comma separated list of ports which will be internally exposed on the spun up docker container.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: of ports which will be -> of ports that will be

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  
+- **External Ports** - This field accepts a comma separated list of ports which will be externally exposed on the host machine.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: of ports which will be -> of ports that will be

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  
+- **Privileged Mode** - This flag if enabled will give the container root capabilities to all devices on the host system. Please be aware that setting this flag can be dangerous, and must only be used in exceptional situations.
+  
+- **Environment Variables (optional)** - This field accepts a comma separated list of environment variables, which will be set inside the container when spun up.
+  
+- **Volume Mount (optional)** - This field accepts a comma separated list of system-to-container file mounts. This allows for the container to access files on the host machine.
+  
+- **Peripheral Device (optional)** - This field accepts a comma separated list of device paths. This parameter allows for devices to be passed though from the host to the container.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: passed **though** from -> passed th**r**ough from

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+![Supported-Container-Configuration]({{ site.baseurl }}/assets/images/builtin/container_orchestrator/container_configuration.png)
+
+## Stopping the container
+
+To stop the container without deleting the component, set the **Enabled** field to **false**, and then press **Apply**. This will delete the running container, but leave this component available for running the container again in the future. If you want to completely remove the container and component, press the **Delete** button to the top right of the screen, and press **Yes** on the confirmation dialog.
+![Stop-Container]({{ site.baseurl }}/assets/images/builtin/container_orchestrator/stop_container.png)
+
+## Container Management Dashboard
+
+The Container Orchestration service also provides the user with a intuitive container dashboard. This dashboard shows all containers running on a gateway, including containers created with Kura and those created manually though the command line interface. To utilize this dashboard the **org.eclipse.container.orchestration.provider** (Docker-API) must be enabled, and the dashboard can be opened by navigating to **Device > Docker Containers**.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: the user with **a** intuitive container dashboard. -> the user with **an** intuitive container dashboard.

Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: those created manually **though** the command line -> those created manually th**r**ough the command line

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+This operation lists all the containers installed in the gateway.
+
+* Request Topic:
+  * **$EDC/account_name/client_id/INVENTORY-V1/GET/containers**
+* Request Payload:
+  * Nothing application-specific beyond the request ID and requester client ID
+* Response Payload:
+  * Installed containers serialized in JSON format
+
+The following JSON message is an example what this request outputs:
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: is an example what this -> is an example **of** what this

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+    }
+    
+The container JSON message is comprised of the following elements:
+    - Name: The name of the docker container.
+    - Version: describes both the container's respective image and tag separated by a colon.
+    - Type: denotes the type of inventory payload
+
+## Start Container
+
+This operation allows to start a container installed on the gateway.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: allows to start -> allows starting

File: _posts/builtin/2022-02-23-container-orchestration-provider.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+* Request Topic
+  * $EDC/account_name/client_id/INVENTORY-V1/EXEC/containers/_stop
+* Request Payload
+  * A JSON object that identifies the target container must be specified in payload body. This payload will be described in the following section.
+* Response Payload
+  *Nothing application specific
+
+## JSON identifier/payload for start and stop requests
+
+The requests for starting and stopping a container require the application to include a JSON object in request payload for selecting the target container. Docker enforces unique container names on a gateway, and thus they can reliably be used as a identifier.
```
Comment (marcellorinaldo) [inline _posts/builtin/2022-02-23-container-orchestration-provider.md]: be used as **a** identifier. -> be used as **an** identifier.

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]

**Comment by GregoryIvo** [comment]
Made the suggested changes. Thank you for the Feedback @marcellorinaldo.

**Review by marcellorinaldo — APPROVED** [review]

**Pull request merged**